packages/database/src/migrations/20210831002757-AddEntitiesRelationToSamoaForPenfaaSamoa-modifies-data.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+import { insertObject, codeToId, generateId, deleteObject } from '../utilities';
+import samoaHierarchyData from './migrationData/20210831002757-AddEntitiesRelationToSamoaForPenfaaSamoa/Entites_Hierarchy_Pen_Faa.json';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const projectCode = 'penfaa_samoa';
+
+const hierarchyNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = '${name}'`);
+  return record?.rows[0].id;
+};
+
+exports.up = async function (db) {
+  const entityHierarchyId = generateId();
+  await insertObject(db, 'entity_hierarchy', {
+    id: entityHierarchyId,
+    name: projectCode,
+    canonical_types: '{country,district,sub_district,school}',
+  });
+
+  Promise.all(
+    samoaHierarchyData.map(async ({ parent_code: parentCode, code }) => {
+      const parentId = await codeToId(db, 'entity', parentCode);
+      const childId = await codeToId(db, 'entity', code);
+      await insertObject(db, 'entity_relation', {
```

Comment:
So to record our conversation - schools have already been imported with sub_district parents - probably best to leave school as canonical (not create entity_relations) so that can be updated by admin panel.

Reply:
and maybe double check that schools don't show up in covid_samoa :)

Reply:
Checked that schools are not showing in my local, but will let testers to check that as well.